src/Peachpie.CodeAnalysis/CodeGen/VariableReference.cs

```diff
@@ -1110,6 +1110,11 @@ public void Store(CodeGenerator cg, TypeSymbol valuetype)
 
         #endregion
 
+        /// <summary>
+        /// Whether to skip passing the parameter value, as it is unnecessary (e.g. unused parameter or only delegation to another method)
+        /// </summary>
+        public bool SkipPass { get; set; }
```

Comment:
it shouldn't be in `VariableReference` ... 

can we just refer to the FlowState (as we do with types)?

Reply:
We can add a bitmask to `FlowContext`. That's still not perfect, as the analysis is actually not performed during the main analysis phase, but maybe better than now?

Reply:
`SkipPass` only makes sense for `SourceParameter` right ?

Reply:
Yes.

Reply:
good, then it makes more sense to make a property in `SourceParameterSymbol` called i.e. `CopyOnPass` ? will be false for *byref*, and some specific types (*string*, *bool*, *long*, *double*, *object*, ..., even nullable) and when your analysis says that.

Reply:
OK, see 96eb30660bd40b4ca997494b0068443601f14761.